### PR TITLE
feat: adding winston and some logs in audits module

### PR DESCRIPTION
**Pull request by zehzo**
image

I am using Winston, the screenshot above shows how the new logger turned out.

image

I preferred to implement Winston because it allows me certain customizations and also enables me to save logs from previous days. I configured it to store logs for up to 14 days.

image

Here is an example of the log showing an error (I’m still going to work on the application’s error management). Focusing on the logs, I believe that what I implemented in the audit controller and service should cover this entity adequately with logs. However, I’m still working on it, and I would like you to review it and give me feedback. If everything looks good, I will implement the logs in the other entities as well.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -7,15 +8,55 @@ import {
...
+        format: winston.format.combine(
+          winston.format.timestamp(),
+          winston.format.json(),
+        ),
+        level: 'info',
+      }),
+    ],
+
+    silent: process.env.NODE_ENV === 'test',
+  });
```
Comment (yurimutti) [inline src/main.ts]: Please, move this code to another file called winston.logger.ts.

and import like this:

```
// imports for Winston logger
import { WinstonModule } from 'nest-winston';
import { instance } from 'logger/winston.logger';
```

Comment (yurimutti) [reply src/main.ts]: Its important show  logs for diferents environments, on develop more readable and more JSON on production.

```
import { createLogger, format, transports } from "winston";

// custom log display format
const customFormat = format.printf(({timestamp, level, stack, message}) => {
    return `${timestamp} - [${level.toUpperCase().padEnd(7)}] - ${stack || message}`
})

const options = {
    file: {
        filename: 'error.log',
        level: 'error'
    },
    console: {
        level: 'silly'
    }
}

// for development environment
const devLogger = {
    format: format.combine(
        format.timestamp(),
        format.errors({stack: true}),
        customFormat
    ),
    transports: [new transports.Console(options.console)]
}

// for production environment
const prodLogger = {
    format: format.combine(
        format.timestamp(),
        format.errors({stack: true}),
        format.json()
    ),
    transports: [
        new transports.File(options.file),
        new transports.File({
            filename: 'combine.log',
            level: 'info'
        })
    ]
}

// export log instance based on the current environment
const instanceLogger = (process.env.NODE_ENV === 'production') ? prodLogger : devLogger

export const instance = createLogger(instanceLogger)

```

File: src/modules/audit/audit.controller.ts [hunk 2]
```diff
@@ -22,6 +23,8 @@ import { UpdateAuditDto, UpdateAuditSchema } from './dtos/update-audit.dto';
 @ApiTags('audits')
 @Controller({ path: 'audits', version: '1' })
 export class AuditController {
+  private readonly logger = new Logger(AuditController.name); // Initialize logger for the controller
```
Comment (yurimutti) [inline src/modules/audit/audit.controller.ts]: Please, can you create a module for this? and use injection for instance:

` private readonly logger: Logger, // instantiate logger`

```
import { Logger, Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import { UserModule } from './user/user.module';

@Module({
  imports: [
    UserModule,
  ],
  providers: [Logger],
})

export class AppModule {}

```

**Review by yurimutti — CHANGES_REQUESTED** [review]
- Move winston setup to another file
- Structure diferent logs per environment
- Use logger as module

File: jest.config.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  transform: {
+    '^.+\\.tsx?$': 'ts-jest',
+  },
+  transformIgnorePatterns: ['/node_modules/'],
+  moduleNameMapper: {
+    '^@/(.*)$': '<rootDir>/src/$1',
+  },
+};
```
Comment (yurimutti) [inline jest.config.ts]: We use this file to jest config -> https://github.com/detrash/recy-api/blob/staging/test/jest.shared.ts

Why you need create this aditional config ?

Comment (zehzo) [reply jest.config.ts]: You're right, it's really not necessary. I was having trouble running the tests due to the logger module implementation, but I ended up resolving it without needing that and didn’t remove it. I'll delete this file.

**Review by yurimutti — APPROVED** [review]

**Pull request merged**